Content.Shared/Delivery/SharedDeliverySystem.cs

```diff
@@ -47,8 +52,62 @@ public override void Initialize()
 
         SubscribeLocalEvent<DeliverySpawnerComponent, ExaminedEvent>(OnSpawnerExamine);
         SubscribeLocalEvent<DeliverySpawnerComponent, GetVerbsEvent<AlternativeVerb>>(OnGetSpawnerVerbs);
+
+        SubscribeLocalEvent<DeliveryComponent, AfterInteractEvent>(OnDeliveryAfterInteract);
+        SubscribeLocalEvent<DeliveryComponent, DeliveryForceUnlockDoAfterEvent>(OnDeliveryForceUnlockDoAfter);
+    }
+
+    #region Force Unlock
+    private void OnDeliveryAfterInteract(Entity<DeliveryComponent> ent, ref AfterInteractEvent args)
+    {
+        if (!args.CanReach || args.Target == null || args.Target == args.User || !ent.Comp.CanForceDeliver)
+            return;
+
+        if (!HasComp<FingerprintComponent>(args.Target.Value))
+            return;
+
+        if (!ent.Comp.IsLocked)
+            return;
+
+        var usedName = _nameModifier.GetBaseName(ent.Owner);
+
+        _popup.PopupPredicted(Loc.GetString("delivery-force-popup-target", ("identity", Identity.Name(args.User, EntityManager))), null, args.Target.Value, args.Target.Value, PopupType.SmallCaution);
+        _popup.PopupPredicted(Loc.GetString("delivery-force-popup-self", ("identity", Identity.Name(args.Target.Value, EntityManager)), ("type", usedName)), null, args.User, args.User);
+
+        var doAfterArgs = new DoAfterArgs(EntityManager, args.User, ent.Comp.ForceDeliverTime, new DeliveryForceUnlockDoAfterEvent(), args.Used, args.Target, args.Used)
+        {
+            NeedHand = true,
+            BreakOnDamage = true,
+            BreakOnMove = true,
+            BreakOnDropItem = true,
+            BreakOnHandChange = true,
+        };
+
+        _doAfterSystem.TryStartDoAfter(doAfterArgs);
+    }
+
+    private void OnDeliveryForceUnlockDoAfter(Entity<DeliveryComponent> ent, ref DeliveryForceUnlockDoAfterEvent args)
+    {
+        if (args.Cancelled || args.Target == null || args.Target == args.User)
+            return;
+
+        if (!_fingerprintReader.IsAllowed(ent.Owner, args.Target.Value))
+        {
+            var usedName = _nameModifier.GetBaseName(ent.Owner);
+            _popup.PopupPredicted(Loc.GetString("delivery-force-popup-self-fail", ("identity", Identity.Name(args.Target.Value, EntityManager)), ("type", usedName)), null, args.User, args.User);
+            return;
+        }
+
+        if (!ent.Comp.IsLocked)
+            return;
+
+        _hands.TryPickupAnyHand(args.Target.Value, ent);
+
+        TryUnlockDelivery(ent, args.Target.Value);
```

Comment:
Desyncing this between tryunlockdelivery (they each do their own fingerprintreader + locked checks) will just cause bugs and I suggest you align the 2 methods.